packages/swing-store/src/swingStore.js

```diff
@@ -82,6 +87,154 @@ export function makeSnapStoreIO() {
  * }} SwingStore
  */
 
+/**
+ * @typedef {[
+ *   key: string,
+ *   value: string|null,
+ * ]} KVPair
+ *
+ * @typedef {object} SwingStoreExporter
+ *
+ * Allows export of data from a swingStore as a fixed view onto the content as
+ * of the most recent commit point at the time the exporter was created.  The
+ * exporter may be used while another SwingStore instance is active for the same
+ * DB, possibly in another thread or process.  It guarantees that regardless of
+ * the concurrent activity of other swingStore instances, the data representing
+ * the commit point will stay consistent and available.
+ *
+ * @property {() => Iterable<KVPair>} getKVData
+ *
+ * Get a full dump of KV data from the swingStore. This represents both the
+ * KVStore (excluding host and local prefixes), as well as any data needed to
+ * validate all artifacts, both current and historical. As such it represents
+ * the root of trust for the application.
+ *
+ * Likely content of validation data (with supporting entries for indexing):
+ * - export.snapshot.${vatID}.${endPos} = ${{ vatID, endPos, hash, uncompressedSize, compressedSize });
+ * - export.lastSnapshot.${vatID} = ${endPos}
+ * - export.transcript.${vatID}.${startPos} = ${{ vatID, startPos, endPos, hash, uncompressedSize, compressedSize }}
+ *
+ * @property {(includeHistorical: boolean) => Iterable<string>} getArtifactNames
+ *
+ * Get a list of name of artifacts available from the swingStore A name returned
+ * by this method guarantees that a call to `getArtifact` on the same exporter
+ * instance will succeed. Options control the filtering of the artifact names
+ * yielded.
+ *
+ * Likely artifact names:
+ * - export.transcript.${vatID}.${startPos}
```

Comment:
I am feeling very queasy about an artifact with a given name mutating over time. I really would much rather have an artifact name represent stable content, which is why I had suggested this should be named `export.transcript.${vatID}.${startPos}.${endPod}` instead.

Reply:
hm, let's talk about it, I think "artifacts have unique name->contents" might imply that the artifacts are so long-lived that they aren't scoped to a particular export, or that they can be preserved from one export to the next, and that feels like it breaks some other assumptions, or at least if that were the case then some other parts of the design would be very different.